transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java

```diff
@@ -484,17 +484,9 @@ private static int nioBuffers(Entry entry, ByteBuf buf, ByteBuffer[] nioBuffers,
     }
 
     private static ByteBuffer[] expandNioBufferArray(ByteBuffer[] array, int neededSpace, int size) {
-        int newCapacity = array.length;
-        do {
-            // double capacity until it is big enough
-            // See https://github.com/netty/netty/issues/1890
-            newCapacity <<= 1;
-
-            if (newCapacity < 0) {
```

Comment:
You've also removed the overflow check?

Reply:
Ya, no real need for it

`newCapacity = Math.max(newCapacity, neededSpace)`

Will cover the case of an overflow.  If newCapacity overflows and goes negative, it will be 'max'-ed out by the neededSpace.